### PR TITLE
Update the minimal vscode engine version and dependencies

### PR DESCRIPTION
**Pull request by aaharu**
update and fix working with the latest version of XO

https://github.com/SamVerschueren/vscode-linter-xo/issues/91

image

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+	"xo": {
+		"ignores": [
+			"!out/**"
+		],
+		"rules": {
+			"@typescript-eslint/no-namespace": "warn",
+			"unicorn/prefer-node-protocol": "off",
+			"unicorn/prevent-abbreviations": "warn"
+		}
+	}
```
Comment (yakov116) [inline client/package.json:20]: Why was this added?

File: server/package.json [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+			"@typescript-eslint/no-implicit-any-catch": "warn",
+			"@typescript-eslint/no-namespace": "warn",
+			"@typescript-eslint/no-unsafe-member-access": "warn",
+			"@typescript-eslint/no-unsafe-assignment": "warn",
+			"@typescript-eslint/no-unsafe-call": "warn",
+			"unicorn/no-abusive-eslint-disable": "warn",
+			"unicorn/prefer-node-protocol": "off",
+			"unicorn/prevent-abbreviations": "warn"
+		}
+	}
```
Comment (yakov116) [inline server/package.json:25]: And this?

**Comment by yakov116** [comment]
@SamVerschueren can you please review?

**Comment by Spence-S** [comment]
Thanks for your time looking into this, since I published my fork and assumed maintenance here we won't be going this direction. If you want to take a stab into any more issues feel free!
